client/src/components/ui/QRScanner.svelte

```diff
@@ -0,0 +1,108 @@
+    let uploadElement = null as HTMLInputElement | null;
+
+    async function startCamera() {
+        camStart = true;
+        await qrScanner?.start();
+    }
+
+    function stopCamera() {
+        camStart = false;
+        return qrScanner?.stop();
+    }
+
+    async function handleFileInput() {
+        assert(uploadElement instanceof HTMLInputElement);
+        assert(uploadElement.files instanceof FileList);
+        const [first] = uploadElement.files;
+        assert(typeof first !== 'undefined');
+        try {
+            const scanData = await QrScanner.scanImage(first, { returnDetailedScanResult: true });
+            if (scanData.data === '') return;
+            maybeId = DocumentSchema.shape.id.parse(scanData.data.trim());
+            dispatch(Events.OnDocumentScan, maybeId);
+        } catch (err) {
+            topToastMessage.enqueue({ title: 'Failed to Find QR code', body: JSON.stringify(err) });
+        }
+        uploadElement.value = '';
+    }
+
+    function setup() {
+        assert(videoElement instanceof HTMLVideoElement);
+
+        return new QrScanner(
+            videoElement,
+            result => {
+                if (result.data === '') return;
+                maybeId = DocumentSchema.shape.id.parse(result.data.trim());
+                stopCamera();
+                dispatch(Events.OnDocumentScan, maybeId);
+            }, {
+                highlightCodeOutline: true,
+                highlightScanRegion: true,
+                onDecodeError: (err) => {
+                    if (err === 'No QR code found') return;
+                    topToastMessage.enqueue({ title: 'Failed to Find QR code', body: JSON.stringify(err) });
+                },
+                maxScansPerSecond: 1,
+                returnDetailedScanResult: true,
+            }
+        );
+    }
+    
+    onMount(() => {
+        // Needs to be here so that the <video> tag is rendered before it is asserted
+        qrScanner = setup();
+    });
+
+    onDestroy(() => {
+        // Otherwise, when the window is closed - browser still thinks you are recording.
+        stopCamera();
+    });
+</script>
+
+<video bind:this={videoElement}><track kind="captions"></video>
+{#await QrScanner.hasCamera()}
+    Loading cameras...
+    {:then hasCamera}
+        {#if hasCamera}
+        <header>
+            {#if camStart}
+                <Button type={ButtonType.Secondary} on:click={() => stopCamera()}> End Capture </Button>
+            {:else}
+                <Button on:click={() => startCamera()}> Start Capture </Button>
+            {/if}
+        </header>
+        {:else}
+            <p> No cameras available. </p>
+        {/if}
+{/await}
+<header>
+    Upload a file with the QR code: <input bind:this={uploadElement} on:change={() => handleFileInput()} type="file" id="file-selector" accept="image/*">
+</header>
+    {#if maybeId === null || maybeId === ''} 
+        <p> No valid QR code detected. </p>
+    {/if}
```

Comment:
Just some more indentation issues.
```suggestion
{#if maybeId === null || maybeId === ''} 
    <p> No valid QR code detected. </p>
{/if}
```